### PR TITLE
[search][search_quality] Convert samples_generation_tool parameters to command line args.

### PR DESCRIPTION
**Pull request by tatiana-yan**

### CODE REVIEW

File: search/search_quality/samples_generation_tool/samples_generation_tool.cpp [hunk 1]
```diff
@@ -38,16 +39,25 @@ using namespace search::search_quality;
...
 double constexpr kMinViewportSizeM = 100.0;
 double constexpr kMaxViewportSizeM = 5000.0;
 size_t constexpr kMaxSamplesPerMwm = 20;
 
+DEFINE_string(data_path, "", "Path to data directory (resources dir).");
+DEFINE_string(mwm_path, "", "Path to mwm files (writable dir).");
+DEFINE_string(out_buildings_path, "buildings.json", "Path to output file for buildings samples.");
+DEFINE_string(out_cafes_path, "cafes.json", "Path to output file for cafes samples.");
+DEFINE_double(max_distance_to_object, kMaxDistanceToObjectM,
+              "Maximal distance from user position  to object (meters).");
```
Comment (mpimenov) [inline search/search_quality/samples_generation_tool/samples_generation_tool.cpp]: extra space after position

File: search/search_quality/samples_generation_tool/samples_generation_tool.cpp [hunk 2]
```diff
@@ -192,20 +205,25 @@ void ModifyHouse(uint8_t lang, string & str)
 
 m2::PointD GenerateNearbyPosition(m2::PointD const & point)
 {
-  uniform_real_distribution<double> dis(-kMaxDistanceToObjectM, kMaxDistanceToObjectM);
+  auto const maxDistance = FLAGS_max_distance_to_object;
+  uniform_real_distribution<double> dis(-1 * maxDistance, maxDistance);
```
Comment (mpimenov) [inline search/search_quality/samples_generation_tool/samples_generation_tool.cpp]: -maxDistance ?

File: search/search_quality/samples_generation_tool/samples_generation_tool.cpp [hunk 3]
```diff
@@ -192,20 +205,25 @@ void ModifyHouse(uint8_t lang, string & str)
...
+  uniform_real_distribution<double> dis(FLAGS_min_viewport_size, FLAGS_max_viewport_size);
   return MercatorBounds::RectByCenterXYAndSizeInMeters(GenerateNearbyPosition(point), dis(g_rng));
 }
 
 bool GetBuildingInfo(FeatureType & ft, search::ReverseGeocoder const & coder, string & street)
 {
-  if (ft.GetHouseNumber().empty())
+  auto const houseNumber = ft.GetHouseNumber();
+  if (houseNumber.empty() || !search::house_numbers::LooksLikeHouseNumber(
+                                 strings::MakeUniString(houseNumber), false /* prefix */))
```
Comment (mpimenov) [inline search/search_quality/samples_generation_tool/samples_generation_tool.cpp]: Maybe add the override
```
bool LooksLikeHouseNumber(std::string const & utf8s, bool isPrefix)
```
?

File: search/search_quality/samples_generation_tool/samples_generation_tool.cpp (hunk 1 above)
Comment (tatiana-yan) [reply search/search_quality/samples_generation_tool/samples_generation_tool.cpp]: done

File: search/search_quality/samples_generation_tool/samples_generation_tool.cpp (hunk 2 above)
Comment (tatiana-yan) [reply search/search_quality/samples_generation_tool/samples_generation_tool.cpp]: done

File: search/search_quality/samples_generation_tool/samples_generation_tool.cpp (hunk 3 above)
Comment (tatiana-yan) [reply search/search_quality/samples_generation_tool/samples_generation_tool.cpp]: done

**Pull request merged**